testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/authentication/ExecutionTest.java

```diff
@@ -312,4 +315,44 @@ public void testClientFlowExecutions() {
         authMgmtResource.deleteFlow(rep.getId());
         assertAdminEvents.assertEvent(REALM_NAME, OperationType.DELETE, AdminEventPaths.authFlowPath(rep.getId()), ResourceType.AUTH_FLOW);
     }
+
+    @Test
+    public void testRequirementsInExecution() {
+        HashMap<String, String> params = new HashMap<>();
+        String newBrowserFlow = "new-exec-flow";
+
+        params.put("newName", newBrowserFlow);
+        try (Response response = authMgmtResource.copy("browser", params)) {
+            assertAdminEvents.assertEvent(REALM_NAME, OperationType.CREATE, AdminEventPaths.authCopyFlowPath("browser"), params, ResourceType.AUTH_FLOW);
+            Assert.assertEquals("Copy flow", 201, response.getStatus());
+        }
+
+        addExecutionCheckReq(newBrowserFlow, UsernameFormFactory.PROVIDER_ID, params);
+        addExecutionCheckReq(newBrowserFlow, WebAuthnAuthenticatorFactory.PROVIDER_ID, params);
+        addExecutionCheckReq(newBrowserFlow, NoCookieFlowRedirectAuthenticatorFactory.PROVIDER_ID, params);
+
+        AuthenticationFlowRepresentation rep = findFlowByAlias(newBrowserFlow, authMgmtResource.getFlows());
+        Assert.assertNotNull(rep);
+        authMgmtResource.deleteFlow(rep.getId());
+        assertAdminEvents.assertEvent(REALM_NAME, OperationType.DELETE, AdminEventPaths.authFlowPath(rep.getId()), ResourceType.AUTH_FLOW);
+    }
+
+    private void addExecutionCheckReq(String flow, String providerID, HashMap<String, String> params) {
+        params.put("provider", providerID);
+        authMgmtResource.addExecution(flow, params);
+        assertAdminEvents.assertEvent(REALM_NAME, OperationType.CREATE, AdminEventPaths.authAddExecutionPath(flow), params, ResourceType.AUTH_EXECUTION);
+
+        List<AuthenticationExecutionInfoRepresentation> executionReps = authMgmtResource.getExecutions(flow);
+        AuthenticationExecutionInfoRepresentation exec = findExecutionByProvider(providerID, executionReps);
+
+        Assert.assertNotNull(exec);
+
+        if (exec.getRequirementChoices().size() == 1)
```

Comment:
I would vote if this method "addExecutionCheckReq" has an additional argument like "expectedRequirement" . Then instead of this if/else, you will just explicitly check if the expected requirement matches with the real execution requirement.

Reply:
Yes, it would be better. Thanks @mposolda